client/src/components/Joystick.tsx

```diff
@@ -0,0 +1,92 @@
+import { Joystick } from 'react-joystick-component'
+
+export interface JoystickMovement {
+  isMoving: boolean
+  direction: Direction
+}
+
+type Direction = {
+  left: boolean
+  right: boolean
+  up: boolean
+  down: boolean
+}
+
+type Props = {
+  onDirectionChange: (arg: JoystickMovement) => void
+}
+
+const angleToDirections = (angle: number, out?: Partial<Direction> | boolean): Direction => {
+  const outObj: Direction = {
+    left: false,
+    right: false,
+    up: false,
+    down: false,
+    ...(typeof out === 'object' ? out : {}),
```

Comment:
this got me a little confused, if we're setting all left, right, up, down to false anyway, is there a reason to merge the input object here?
I would recommend getting rid of the second arg here and at the callsite.

Reply:
Sorry about this, my bad.. I was doing some tests and I forgot refactor it 😅 I'll change right now 👍 You'r right, only 1 argument it's needed